File: src/main/java/com/dingdo/common/exception/BusinessException.java
package com.dingdo.common.exception;

/**
 * @author slamacraft
 * @version 1.0
 * @date 2020/9/18 16:24
 * @since JDK 1.8
 */
public class BusinessException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    private String message;

    public BusinessException(String message) {
        super(message);
        this.message = message;
    }

    public BusinessException(String message, Throwable e) {
        super(message, e);
        this.message = message;
    }

    public String getmessage() {
        return message;
    }

    public void setmessage(String message) {
        this.message = message;
    }

}

File: src/main/scala/com/dingdo/common/exceptions/BusinessException.scala
package com.dingdo.common.exceptions

trait BusinessException extends RuntimeException {
  var code: String = "500"
}

object BusinessException{
  def apply(msg:String):BusinessException = new RuntimeException(msg) with BusinessException
  def apply(code:String, msg:String): BusinessException = {
    val exception = new RuntimeException(msg) with BusinessException
    exception.code = code
    exception
  }
  def apply(msg:String, e:Throwable):BusinessException = new RuntimeException(msg, e) with BusinessException
}
